ros/processor/inventory_events_consumer.py

```diff
@@ -103,27 +108,79 @@ def host_create_update_events(self, msg):
 
     def process_system_details(self, msg):
         """ Store new system information (stale, stale_warning timestamp) and return internal DB id"""
         host = msg['host']
+        insights_report_tar = self._download_report(msg['platform_metadata']['url'])
+        performance_record = self._extract_performance_record(insights_report_tar)
+        if performance_record:
+            performance_score = self._calculate_performance_score(performance_record)
 
-        account = get_or_create(
-            db.session, RhAccount, 'account',
-            account=host['account']
-        )
-
-        system = get_or_create(
-            db.session, System, 'inventory_id',
-            account_id=account.id,
-            inventory_id=host['id'],
-            display_name=host['display_name'],
-            fqdn=host['fqdn'],
-            cloud_provider=host['system_profile']['cloud_provider']
-        )
-        # Commit changes
-        db.session.commit()
-        LOG.warning("Refreshed system %s (%s) belonging to account: %s (%s)",
-                    system.inventory_id, system.id, account.account, account.id)
+            account = get_or_create(
+                db.session, RhAccount, 'account',
+                account=host['account']
+            )
+
+            system = get_or_create(
+                db.session, System, 'inventory_id',
+                account_id=account.id,
+                inventory_id=host['id'],
+                display_name=host['display_name'],
+                fqdn=host['fqdn'],
+                cloud_provider=host['system_profile']['cloud_provider']
+            )
+
+            get_or_create(
+                db.session, PerformanceProfile, ['system_id', 'report_date'],
+                system_id=system.id,
+                performance_record=performance_record,
+                performance_score=performance_score,
+                report_date=date.today()
+            )
+
+            # Commit changes
+            db.session.commit()
+            LOG.warning("Refreshed system %s (%s) belonging to account: %s (%s)",
+                        system.inventory_id, system.id, account.account, account.id)
+
+    def _download_report(self, report_url):
+        download_response = requests.get(report_url)
+        if download_response.status_code != HTTPStatus.OK:
+            print("Unable to download the report")
+        return download_response.content
+
+    def _extract_performance_record(self, report_tar):
+        tar = tarfile.open(fileobj=BytesIO(report_tar), mode='r:*')
+        files = tar.getmembers()
+        performance_record = {}
+        spec_count = 0
+        for file in files:
+            if 'data/insights_commands/pmlogsummary' in file.name:
+                spec_count += 1
+                extracted_pmlogsummary_file = tar.extractfile(file)
+                record_string = extracted_pmlogsummary_file.read().decode('utf-8')
+                records = record_string.split('\n')
```

Comment:
It seems similar logic needed for each file read, it will be better to extract this in separate method & use it.